test_BaseRClassifier_with_sample_weights fails with xgboost 1.6

**Describe the bug**
The test_BaseRClassifier_with_sample_weights fails for me running out of a fresh clone, with no changes, when using xgboost 1.6.

**To Reproduce**
Download a fresh clone of the repo and follow the test steps in https://github.com/uber/causalml/blob/3d59609952b53f8f220517b34257608dda07845c/CONTRIBUTING.md#test-wrench.

When installing dependencies, pip automatically chooses the latest xgboost 1.6. If I manually specify xgboost 1.5.2, all tests pass on my system.

**Expected behavior**
Tests pass instead of failing.

**Environment (please complete the following information):**
 - OS: Arch Linux
 - Python Version: 3.10
 - Versions of Major Dependencies (`pandas`, `scikit-learn`, `cython`): `pandas==1.4.2`, `scikit-learn==1.0.2`, `cython==0.29.28`]

**Additional context**
Error from the failing test (seems like it happens in xgboost internals while training the model):
```
        # Check if XGBRRegressor successfully produces treatment effect estimation
        # when sample_weight is passed
        uplift_model = XGBRRegressor()
>       uplift_model.fit(
            X=df_train[x_names].values,
            p=df_train["propensity_score"].values,
            treatment=df_train["treatment_group_key"].values,
            y=df_train[CONVERSION].values,
            sample_weight=df_train["sample_weights"],
        )

tests/test_meta_learners.py:846: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
causalml/inference/meta/rlearner.py:646: in fit
    self.models_tau[group].fit(
../cml-venv/lib/python3.10/site-packages/xgboost/core.py:532: in inner_f
    return f(**kwargs)
../cml-venv/lib/python3.10/site-packages/xgboost/sklearn.py:961: in fit
    self._Booster = train(
../cml-venv/lib/python3.10/site-packages/xgboost/core.py:532: in inner_f
    return f(**kwargs)
../cml-venv/lib/python3.10/site-packages/xgboost/training.py:182: in train
    if cb_container.after_iteration(bst, i, dtrain, evals):
../cml-venv/lib/python3.10/site-packages/xgboost/callback.py:239: in after_iteration
    score: str = model.eval_set(evals, epoch, self.metric, self._output_margin)
../cml-venv/lib/python3.10/site-packages/xgboost/core.py:1803: in eval_set
    _check_call(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

ret = -1

    def _check_call(ret: int) -> None:
        """Check the return value of C API call
    
        This function will raise exception when error occurs.
        Wrap every API call with this function
    
        Parameters
        ----------
        ret : int
            return value from API calls
        """
        if ret != 0:
>           raise XGBoostError(py_str(_LIB.XGBGetLastError()))
E           xgboost.core.XGBoostError: [21:51:41] ../src/metric/auc.cc:324: Check failed: auc <= local_area (224.434 vs. 196.9) : 
E           Stack trace:
E             [bt] (0) .../cml-venv/lib/python3.10/site-packages/xgboost/lib/libxgboost.so(+0x236379) [0x7fd880836379]
E             [bt] (1) .../cml-venv/lib/python3.10/site-packages/xgboost/lib/libxgboost.so(+0x2509e7) [0x7fd8808509e7]
E             [bt] (2) .../cml-venv/lib/python3.10/site-packages/xgboost/lib/libxgboost.so(+0x2042ff) [0x7fd8808042ff]
E             [bt] (3) .../cml-venv/lib/python3.10/site-packages/xgboost/lib/libxgboost.so(XGBoosterEvalOneIter+0x266) [0x7fd8806ab796]
E             [bt] (4) /usr/lib/libffi.so.8(+0x6d4a) [0x7fd8ade2fd4a]
E             [bt] (5) /usr/lib/libffi.so.8(+0x6267) [0x7fd8ade2f267]
E             [bt] (6) /usr/lib/python3.10/lib-dynload/_ctypes.cpython-310-x86_64-linux-gnu.so(+0x7e9e) [0x7fd8aa898e9e]
E             [bt] (7) /usr/lib/python3.10/lib-dynload/_ctypes.cpython-310-x86_64-linux-gnu.so(+0x75fc) [0x7fd8aa8985fc]
E             [bt] (8) /usr/lib/libpython3.10.so.1.0(_PyObject_MakeTpCall+0x2ab) [0x7fd8ae8aa7cb]

../cml-venv/lib/python3.10/site-packages/xgboost/core.py:203: XGBoostError
```

--- a/causalml/inference/meta/rlearner.py
+++ b/causalml/inference/meta/rlearner.py
@@ -516,7 +516,7 @@ class XGBRRegressor(BaseRRegressor):
         early_stopping=True,
         test_size=0.3,
         early_stopping_rounds=30,
-        effect_learner_objective="rank:pairwise",
+        effect_learner_objective="reg:squarederror",
         effect_learner_n_estimators=500,
         random_state=42,
         *args,
@@ -531,7 +531,7 @@ class XGBRRegressor(BaseRRegressor):
             early_stopping_rounds (int, optional): validation metric needs to improve at least once in every
                                                    early_stopping_rounds round(s) to continue training
             effect_learner_objective (str, optional): the learning objective for the effect learner
-                                                      (default = 'rank:pairwise')
+                                                      (default = 'reg:squarederror')
             effect_learner_n_estimators (int, optional): number of trees to fit for the effect learner (default = 500)
         """
 

--- a/tests/test_meta_learners.py
+++ b/tests/test_meta_learners.py
@@ -840,18 +840,24 @@ def test_BaseRClassifier_with_sample_weights(generate_classification_data):
     # higher than it would be under random targeting
     assert cumgain["tau_pred"].sum() > cumgain["Random"].sum()
 
+
+def test_XGBRegressor_with_sample_weights(generate_regression_data):
+    y, X, treatment, tau, b, e = generate_regression_data()
+
+    weights = np.random.rand(y.shape[0])
+
     # Check if XGBRRegressor successfully produces treatment effect estimation
     # when sample_weight is passed
     uplift_model = XGBRRegressor()
     uplift_model.fit(
-        X=df_train[x_names].values,
-        p=df_train["propensity_score"].values,
-        treatment=df_train["treatment_group_key"].values,
-        y=df_train[CONVERSION].values,
-        sample_weight=df_train["sample_weights"],
+        X=X,
+        p=e,
+        treatment=treatment,
+        y=y,
+        sample_weight=weights,
     )
-    tau_pred = uplift_model.predict(X=df_test[x_names].values)
-    assert len(tau_pred) == len(df_test["sample_weights"].values)
+    tau_pred = uplift_model.predict(X=X)
+    assert len(tau_pred) == len(weights)
 
 
 def test_pandas_input(generate_regression_data):

